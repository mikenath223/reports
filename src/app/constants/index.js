export const pieColors = ["#A259FF", "#F24E1E", "#FFC107", "#6497B1"];
