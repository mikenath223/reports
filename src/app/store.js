import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import projectsReducer from "./slices/projectsSlice";
import formReducer from "./slices/formSlice";
import userReducer from "./slices/userSlice";
import gatewayReducer from "./slices/gatewaySlice";
import reportsReducer from "./slices/reportsSlice";

const middlewares = [];
if (process.env.NODE_ENV === `development`) {
  middlewares.push(logger);
}

export const store = configureStore({
  reducer: {
    projects: projectsReducer,
    form: formReducer,
    users: userReducer,
    gateways: gatewayReducer,
    reports: reportsReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(middlewares),
});
