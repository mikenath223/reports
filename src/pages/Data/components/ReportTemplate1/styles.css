body {
  overflow-x: hidden;
}

.reports-wrap {
  margin-block: 30px;
}

.reports-template {
  background-color: #f1fafe;
  border-radius: 10px;
  padding: 20px 30px;
}

td,
th,
tr,
table {
  border: 0 !important;
  border-spacing: 0 !important;
}

td {
  text-align: center;
}

.date {
  text-align: left;
  padding-left: 25px;
}

.amount {
  text-align: right;
  padding-right: 25px;
}

table {
  width: 100%;
  margin-top: 10px;
}

.accordion {
  margin-block: 10px;
}

.panel-header {
  display: flex;
  justify-content: space-between;
  padding-block: 10px;
  width: 100%;
  align-items: center;
}

.title {
  color: #011f4b;
  font-size: 16px;
  font-weight: 700;
}

.MuiAccordion-root {
  margin-top: 16px !important;
}

.MuiAccordion-root,
.MuiButtonBase-root {
  box-shadow: none !important;
  border: none !important;
  border-radius: 10px !important;
}

.MuiButtonBase-root {
  background-color: #fff !important;
  min-height: unset !important;
}

.MuiAccordion-root::before {
  display: none !important;
}

.MuiAccordionSummary-content {
  margin: 8px 0 !important;
}

tr {
  background-color: #f1fafe;
}

th,
td {
  color: #011f4b;
  font-weight: 400;
  font-size: 16px;
  padding: 7px;
}

thead tr,
tbody tr:nth-child(even) {
  background-color: #fff;
}

.MuiCollapse-wrapper,
.grand-total,
.MuiAccordion-root {
  background-color: #f1fafe !important;
}

.grand-total {
  border-radius: 10px;
  margin-top: 20px;
  padding: 20px;
}

.grand-total span {
  color: #011f4b;
  font-size: 16px;
  font-weight: 700;
}
