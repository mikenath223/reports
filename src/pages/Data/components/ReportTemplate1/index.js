import { formatDate } from "app/utils/formatDate";
import Collapsible from "components/Collapsible";
import { useState } from "react";
import { useSelector } from "react-redux";
import PropTypes from "prop-types";
import "./styles.css";

export default function ReportTemplate1({ title, reports }) {
  const gateways = useSelector((state) => state.gateways);
  const [expanded, setExpanded] = useState(
    Object.keys(reports)?.reverse()?.[0]
  );

  const grandTotal = Object.values(reports).reduce((a, c) => a + c.total, 0);

  const renderProject = (projectTitle, i) => {
    const data = reports[projectTitle]?.data;
    return (
      <Collapsible
        key={projectTitle}
        expanded={expanded === projectTitle}
        setExpanded={(event, newExpanded) =>
          setExpanded(newExpanded ? projectTitle : false)
        }
        header={
          <div className="panel-header">
            <h2 className="title">{projectTitle}</h2>
            <span className="title">
              TOTAL:{" "}
              {new Intl.NumberFormat("en-IN").format(
                reports[projectTitle]?.total
              )}{" "}
              USD
            </span>
          </div>
        }
      >
        <table className="table">
          <thead>
            <tr>
              <th className="date">Date</th>
              <th>Gateway</th>
              <th>Transaction ID</th>
              <th className="amount">Amount</th>
            </tr>
          </thead>
          <tbody>
            {data?.map((project) => {
              const gateway = gateways?.data?.find(
                (item) => item.gatewayId === project.gatewayId
              );
              return (
                <tr key={project.amount}>
                  <td className="date">{formatDate(project?.created)}</td>
                  <td>{gateway?.name}</td>
                  <td>{project?.paymentId}</td>
                  <td className="amount">{project?.amount} USD</td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </Collapsible>
    );
  };
  return (
    <section className="reports-wrap">
      <div className="reports-template">
        <h1>{title}</h1>
        <div>
          {Object.keys(reports || {})
            ?.reverse()
            ?.map((projectTitle, i) => renderProject(projectTitle, i))}
        </div>
      </div>
      <div className="grand-total">
        <span>
          TOTAL: {new Intl.NumberFormat("ja-JP").format(Math.round(grandTotal))}
        </span>
      </div>
    </section>
  );
}

ReportTemplate1.propTypes = {
  title: PropTypes.string,
  reports: PropTypes.shape({
    data: PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    ),
    total: PropTypes.number,
  }),
};
