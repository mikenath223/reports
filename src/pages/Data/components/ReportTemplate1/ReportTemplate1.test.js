import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "app/store";
import ReportTemplate1 from ".";

describe("ReportTemplate1", function () {
  it("should match snapshot", () => {
    const props = {
      title: "Project 1",
      reports: {
        data: [],
        total: 0,
      },
    };
    const { container } = render(
      <Provider store={store}>
        <ReportTemplate1 {...props} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
