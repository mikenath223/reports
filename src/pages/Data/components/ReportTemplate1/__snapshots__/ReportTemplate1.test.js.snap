// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportTemplate1 should match snapshot 1`] = `
<div>
  <section
    class="reports-wrap"
  >
    <div
      class="reports-template"
    >
      <h1>
        Project 1
      </h1>
      <div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded Mui-expanded MuiAccordion-gutters accordion css-1elwnq4-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-controls="panel1a-content"
            aria-expanded="true"
            class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
            id="panel1a-header"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
            >
              <div
                class="panel-header"
              >
                <h2
                  class="title"
                >
                  total
                </h2>
                <span
                  class="title"
                >
                  TOTAL:
                   
                  NaN
                   
                  USD
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  aria-labelledby="panel1a-header"
                  class="MuiAccordion-region"
                  id="panel1a-content"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                  >
                    <table
                      class="table"
                    >
                      <thead>
                        <tr>
                          <th
                            class="date"
                          >
                            Date
                          </th>
                          <th>
                            Gateway
                          </th>
                          <th>
                            Transaction ID
                          </th>
                          <th
                            class="amount"
                          >
                            Amount
                          </th>
                        </tr>
                      </thead>
                      <tbody />
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters accordion css-1elwnq4-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-controls="panel1a-content"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
            id="panel1a-header"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
            >
              <div
                class="panel-header"
              >
                <h2
                  class="title"
                >
                  data
                </h2>
                <span
                  class="title"
                >
                  TOTAL:
                   
                  NaN
                   
                  USD
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  aria-labelledby="panel1a-header"
                  class="MuiAccordion-region"
                  id="panel1a-content"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                  >
                    <table
                      class="table"
                    >
                      <thead>
                        <tr>
                          <th
                            class="date"
                          >
                            Date
                          </th>
                          <th>
                            Gateway
                          </th>
                          <th>
                            Transaction ID
                          </th>
                          <th
                            class="amount"
                          >
                            Amount
                          </th>
                        </tr>
                      </thead>
                      <tbody />
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grand-total"
    >
      <span>
        TOTAL: 
        NaN
      </span>
    </div>
  </section>
</div>
`;
